professor2.spec

```diff
@@ -22,13 +23,22 @@ sed -i -e 's|-std=c[+][+]11|-std=c++%{cms_cxx_standard}|' pyext/setup.py
 # Same for Makefile
 grep -q 'CXXSTD := c[+][+]11' Makefile
 sed -i -e 's|CXXSTD := c[+][+]11|CXXSTD := c++%{cms_cxx_standard}|' Makefile
+sed -i -e "s|pip install -vv|pip install --target ../${PYTHON3_LIB_SITE_PACKAGES} -vv|" Makefile
 
-%define build_flags CPPFLAGS=-I${EIGEN_ROOT}/include/eigen3 PYTHON=$(which python3) PROF_VERSION=%{realversion} PYTHONPATH=./${PYTHON3_LIB_SITE_PACKAGES}
+# Update Python2 to Python3
+for i in `ls bin`
+do
+   echo "bin/${i}"
+   sed -i -e 's|/usr/bin/env python|/usr/bin/env python3|' "bin/${i}"
+done
 
+%define build_flags CPPFLAGS=-I${EIGEN_ROOT}/include/eigen3 PYTHON=$(which python3) PROF_VERSION=%{realversion} PYTHONPATH=./${PYTHON3_LIB_SITE_PACKAGES}
 %build
-make %{build_flags}
+make PREFIX=%{i} %{build_flags}
 
 %install
 make install PREFIX=%{i} %{build_flags}
-mv %{i}/${PYTHON3_LIB_SITE_PACKAGES}/%{n}-%{realversion}*.egg %{i}/${PYTHON3_LIB_SITE_PACKAGES}/%{n}
+mv %{i}/${PYTHON3_LIB_SITE_PACKAGES}/%{n}-%{realversion}*.dist* %{i}/${PYTHON3_LIB_SITE_PACKAGES}/%{n}
```

Comment:
Just to check with the usage of `%{n}`, is the python directory really `professor2-2.4.2` not `professor-2.4.2`?

As prof1 got deprecated we may consider renaming the whole package to just `professor` now? (question to cmsdist team)

Reply:
Yes I did checked that (I got stuck by the egg file for a long while!)

Reply:
In addition, the professor executables are explicitly importing `professor2` (e.g. https://gitlab.com/hepcedar/professor/-/blob/main/bin/prof2-I?ref_type=heads#L31), so we might want to keep the name without changing all those importing.